internal/acctest/administrativeaccount/administrative_account_test.go

```diff
@@ -2,111 +2,229 @@ package acctest_test
-				Config: testAccAdministrativeAccount(resourceName, updatedResourceModel),
-				Check:  testAccCheckExpectedAdministrativeAccountAttributes(updatedResourceModel),
-			},
-			{
-				// Test importing the resource
-				Config:                  testAccAdministrativeAccount(resourceName, updatedResourceModel),
-				ResourceName:            "pingfederate_administrative_account." + resourceName,
-				ImportStateId:           initialResourceModel.stateId,
-				ImportState:             true,
-				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"password"},
-			},
-		},
-	})
+	return builder.String()
 }
 
 func testAccAdministrativeAccount(resourceName string, resourceModel administrativeAccountResourceModel) string {
 	return fmt.Sprintf(`
 resource "pingfederate_administrative_account" "%[1]s" {
-  active      = %[2]t
-  description = "%[3]s"
-  roles       = %[4]s
-  password    = "%[5]s"
-  username    = "%[6]s"
+	%[2]s
 }
 
 data "pingfederate_administrative_account" "%[1]s" {
   id = pingfederate_administrative_account.%[1]s.username
-}`, resourceName,
-		resourceModel.active,
-		resourceModel.description,
-		acctest.StringSliceToTerraformString(roles),
-		password,
-		username,
+}`,
+		resourceName,
+		hcl(resourceModel.administrativeAccount),
 	)
 }
 
 // Test that the expected attributes are set on the PingFederate server
-func testAccCheckExpectedAdministrativeAccountAttributes(config administrativeAccountResourceModel) resource.TestCheckFunc {
+func testAccCheckExpectedInitialAdministrativeAccountAttributes(config administrativeAccountResourceModel) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		resourceType := "AdministrativeAccount"
 		testClient := acctest.TestClient()
 		ctx := acctest.TestBasicAuthContext()
-		response, _, err := testClient.AdministrativeAccountsAPI.GetAccount(ctx, config.stateId).Execute()
+		response, _, err := testClient.AdministrativeAccountsAPI.GetAccount(ctx, config.administrativeAccount.Username).Execute()
 		if err != nil {
 			return err
 		}
+		resourceType := "AdministrativeAccount"
 		// Verify that attributes have expected values
-		err = acctest.TestAttributesMatchBool(resourceType, &config.stateId, "active",
-			config.active, *response.Active)
+		err = acctest.TestAttributesMatchBool(resourceType, &config.administrativeAccount.Username, "active",
+			*config.administrativeAccount.Active, response.GetActive())
 		if err != nil {
 			return err
 		}
-		err = acctest.TestAttributesMatchString(resourceType, &config.stateId, "description",
-			config.description, *response.Description)
+
+		err = acctest.TestAttributesMatchStringSlice(resourceType, &config.administrativeAccount.Username, "roles",
+			config.administrativeAccount.Roles, response.Roles)
 		if err != nil {
 			return err
 		}
+		return nil
+	}
+}
 
+func testAccCheckExpectedUpdatedAdministrativeAccountAttributes(config administrativeAccountResourceModel) resource.TestCheckFunc {
```

Comment:
nit: you could combine the two check methods, using nil checks to skip the ones that aren't set in the config, to avoid some of the duplication. Up to you.

Reply:
this is now combined into one method